dbms/src/Interpreters/Join.cpp

```diff
@@ -2444,6 +2752,14 @@ void Join::finishOneBuild()
     --active_build_concurrency;
     if (active_build_concurrency == 0)
     {
+        if (isEnableSpill())
+        {
+            if (hasPartitionSpilled())
+            {
+                trySpillBuildPartitions(true);
+                tryMarkBuildSpillFinish();
+            }
+        }
         workAfterBuildFinish();
```

Comment:
I think we can move the `trySpillxxx` into `workAfterBuildFinish`

Reply:
> I think we can move the `trySpillxxx` into `workAfterBuildFinish`

done